pkg/modules/inputs/trait/ops_rule.go

```diff
@@ -1,5 +1,45 @@
 package trait
 
+import (
+	"errors"
+	"strconv"
+
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	workspaceapi "kusionstack.io/kusion/pkg/apis/workspace"
+)
+
 type OpsRule struct {
 	MaxUnavailable string `json:"maxUnavailable,omitempty" yaml:"maxUnavailable,omitempty"`
 }
+
+const (
+	OpsRuleConst        = "opsRule"
+	MaxUnavailableConst = "maxUnavailable"
+)
+
+func GetMaxUnavailable(opsRule *OpsRule, workspaceConfig map[string]workspaceapi.GenericConfig) (intstr.IntOrString, error) {
+	var maxUnavailable intstr.IntOrString
+	if opsRule != nil {
+		maxUnavailable = intstr.Parse(opsRule.MaxUnavailable)
+	} else {
+		// An example of opsRule config in workspaceConfig
+		// opsRule:
+		//   maxUnavailable: 1 # or 10%
+		if workspaceConfig[OpsRuleConst] == nil || workspaceConfig[OpsRuleConst][MaxUnavailableConst] == nil {
+			return intstr.IntOrString{}, nil
+		}
+		var wsValue string
+		wsValue, isString := workspaceConfig[OpsRuleConst][MaxUnavailableConst].(string)
+		if !isString {
+			temp, isInt := workspaceConfig[OpsRuleConst][MaxUnavailableConst].(int)
+			if isInt {
+				wsValue = strconv.Itoa(temp)
+			} else {
+				return intstr.IntOrString{}, errors.New("illegal workspace config. opsRule.maxUnavailable in the workspace config is not string or int")
```

Comment:
there is space in `config. opsRule.maxUnavailable `

Reply:
They are two sentences. I add a space deliberately...